tests/integration/Settings_API/SettingTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+use SkyVerge\WooCommerce\PluginFramework\v5_6_1\Settings_API\Setting;
+
+class SettingTest extends \Codeception\TestCase\WPTestCase {
+
+
+	/** @var \IntegrationTester */
+	protected $tester;
+
+
+	protected function _before() {
+
+	}
+
+
+	protected function _after() {
+
+	}
+
+
+	/** Tests *********************************************************************************************************/
+
+
+	/**
+	 * @see Setting::validate_value()
+	 *
+	 * @param mixed $value value to pass to method
+	 * @param string $type setting type
+	 * @param bool $type whether the value should be considered valid or not
+	 *
+	 * @dataProvider provider_validate_value
+	 * */
+	public function test_validate_value( $value, $type, $expected ) {
+
+		$setting = new Setting();
+		$setting->set_type( $type );
+
+		$this->assertSame( $expected, $setting->validate_value( $value ) );
+	}
+
+
+	/**
+	 * Provider for test_validate_value()
+	 *
+	 * @return array
+	 */
+	public function provider_validate_value() {
+
+		require_once( 'woocommerce/Settings_API/Setting.php' );
+
+		return [
+			[ 'example', Setting::TYPE_STRING, true ],
+			[ 3.1415926, Setting::TYPE_STRING, false ],
+
+			[ 'https://skyverge.com/', Setting::TYPE_URL, true ],
+			[ 'file:///tmp/', Setting::TYPE_URL, false ],
+			[ 'example', Setting::TYPE_URL, false ],
+
+			[ 'test@example.com', Setting::TYPE_EMAIL, true ],
+			[ 'not-an-email.com', Setting::TYPE_EMAIL, false ],
+			[ '', Setting::TYPE_EMAIL, false ],
+
+			[ 1729, Setting::TYPE_INTEGER, true ],
+			[ 'hi', Setting::TYPE_INTEGER, false ],
+
+			[ 3.14, Setting::TYPE_FLOAT, true ],
+			[ 3000, Setting::TYPE_FLOAT, false ],
```

Comment:
Should we not accept integers as options/value in a float setting?

Reply:
I think we can change that. I checked and if we were using type hints and `declare(strict_types=1);`, float parameters would accept both int and float values, and integer parameters would accept only ints.

Without `declare(strict_types=1);` both float and int parameters accept int, float, and numeric strings.

In either case it seems we should accept integers as options/values for float settings too. I can't make that update.

Reply:
You can't? 😛

Reply:
lol. For some reason I keep switching positives and negatives when it matters most :/